src/mom5/ocean_core/ocean_sbc.F90

```diff
@@ -4330,6 +4339,18 @@ subroutine flux_adjust(Time, T_diag, Dens, Ext_mode, T_prog, Velocity, river, me
               enddo
           endif
 
+          ! Restrict outgoing salt flux (ice being formed) in cases where salinity falls below
+          ! ocean_ice_salt_limit. If zero_net_salt_restore=.true. then the mismatch will be
+          ! spread globally by the following block of code.
+
+          do j=jsc,jec
+             do i=isc,iec
+                if (T_prog(index_salt)%stf(i,j) < 0.0 .and. T_prog(index_salt)%field(i,j,1,taum1) < ocean_ice_salt_limit * 1000.0) then
```

Comment:
Also even with zero bias it seems the combination of lines 4349, 4362, 4368 will give a T_prog(index_salt)%stf(i,j) flux in line 4368 that is slightly negative (ie the global total flux from line 4362), not zero. So it reduces the problem by a huge amount (the ratio of cells satisfying line 4348 to total ocean area) but doesn't perfectly eliminate the negative flux. But hopefully that would slow the reduction of salinity sufficiently to avoid it going <0.

Reply:
If salinity <0 then we've already entered the twilight zone and the model should have already crashed for any realistic equation of state when it tries to calculate density. Putting a test around the loops is harmless though and probably makes it a bit clearer as to what is going on.

Yes, you're absolutely correct, it can still have a slightly negative flux if the restoration is too weak. That's why I suggest setting the limit to be a bit larger than the concentration of salt in ice.  The net flux into a very fresh cell is equal to flux_restore(original)+flx_restore_total=flux_restore(original)+flx_restore_total(original)+stf_masked_global_mean. The first term is positive and should be large if we're very fresh. The second could be positive or negative but should be tiny. The third is negative but should also be small. I'd estimate 5 orders of magnitude smaller than the first term.